# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/FontFaceImpl.cpp
# Commit: 1f5b0b087930
# Full Hash: 1f5b0b0879309d73ed6b0a03bb8872199e2f2162
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2022-06-29 03:44:30
# Regressor Bug: 1771493
# File Overlap Count: 1
# Description:
#   Bug 1771493 - Part 2. Merge FontFaceSetImpl and gfxUserFontSet into a single class. r=emilio
#   
#   This patch merges FontFaceSetImpl and gfxUserFontSet into the same
#   class. This reduces the level of indirection and in theory makes it
#   easier to manage future threading issues.
# ==============================================================================

diff -r aa5527055ad1 -r 1f5b0b087930 layout/style/FontFaceImpl.cpp
--- a/layout/style/FontFaceImpl.cpp	Tue Jun 28 15:58:09 2022 +0000
+++ b/layout/style/FontFaceImpl.cpp	Tue Jun 28 15:58:10 2022 +0000
@@ -517,10 +517,9 @@
   if (mUserFontEntry) {
     mUserFontEntry->mFontFaces.AppendElement(this);
 
-    MOZ_ASSERT(
-        mUserFontEntry->GetUserFontSet() == mFontFaceSet->GetUserFontSet(),
-        "user font entry must be associated with the same user font set "
-        "as the FontFace");
+    MOZ_ASSERT(mUserFontEntry->GetUserFontSet() == mFontFaceSet,
+               "user font entry must be associated with the same user font set "
+               "as the FontFace");
 
     // Our newly assigned user font entry might be in the process of or
     // finished loading, so set our status accordingly.  But only do so
@@ -744,10 +743,10 @@
 
   for (FontFaceImpl* f : mFontFaces) {
     if (f->mInFontFaceSet) {
-      aResult.AppendElement(f->mFontFaceSet->GetUserFontSet());
+      aResult.AppendElement(f->mFontFaceSet);
     }
     for (FontFaceSetImpl* s : f->mOtherFontFaceSets) {
-      aResult.AppendElement(s->GetUserFontSet());
+      aResult.AppendElement(s);
     }
   }
 